src/main/kotlin/sliv/tool/importer/ProjectParser.kt

```diff
@@ -1,46 +1,110 @@
 package sliv.tool.importer
 
 import javafx.collections.FXCollections
+import javafx.scene.control.Alert
 import javafx.scene.control.TreeItem
+import javafx.scene.image.Image
+import javafx.scene.image.ImageView
+import sliv.tool.parsers.planes.ImagePlanesParser.extractUIDs
+import sliv.tool.project.model.*
+import tornadofx.alert
 import java.io.File
+import kotlin.io.path.Path
 
 object ProjectParser {
-    const val IMAGE_DIRECTORY_NAME = "images"
+    private const val IMAGE_DIRECTORY_NAME = "images"
 
-    val files = FXCollections.observableHashMap<Long, MutableList<OutputInfo>>()
-    val images = FXCollections.observableArrayList<ImageInfo>()
-
-    fun createTreeWithFiles(path: String?, tree: TreeItem<String>): TreeItem<String> {
-        if (path == null) {
-            return tree
+    private fun selectKind(nameDirectory: String, n: Int): LayerKind {
+        return when (nameDirectory.substring(n + 1)) {
+            "keypoint" -> LayerKind.KEYPOINT
+            "plane" -> LayerKind.PLANE
+            "line" -> LayerKind.LINE
+            else -> {
+                throw Exception()
+            }
         }
-        var directory = File(path)
+    }
+
+    fun parseDirectory(path: String): Project {
+        val directory = File(path)
+        val images = FXCollections.observableHashMap<Long, String>()
+        val landmarks = FXCollections.observableHashMap<Long, MutableList<LandmarkFile>>()
+        val frames = FXCollections.observableArrayList<ProjectFrame>()
+        val layers = FXCollections.observableArrayList<ProjectLayer>()
 
         for (folder in directory.listFiles()) {
-            if (folder.name == IMAGE_DIRECTORY_NAME) {
-                images.addAll(folder.listFiles().map {
-                    ImageInfo(it.nameWithoutExtension, it.absolutePath)
-                })
+            if (folder.name != IMAGE_DIRECTORY_NAME) {
+                val nameDirectory = folder.name
+                val n = nameDirectory.lastIndexOf("_")
+                val name = nameDirectory.substring(0, n)
+                try {
+                    val kind = selectKind(nameDirectory, n)
+                    layers.add(ProjectLayer(kind, name))
+                } catch (e: Exception) {
+                    alert(
+                        Alert.AlertType.WARNING,
+                        "The directory $nameDirectory is skipped, because name of folder doesn't contain correct kind name."
+                    )
+                    continue
+                }
+
+                folder.listFiles().forEach {
+                    val imageName = it.nameWithoutExtension
+                    try {
+                        val longName = imageName.toLong()
+                        landmarks.putIfAbsent(
```

Comment:
will it save frames from two algos here? were landmarks are cleared?

Reply:
No, it won't save. I was searching for a method without using if, but I never found it. And I forgot to change that